Scheduler date/time format issue with localized sites.

mean2me opened the pull request:
This pull request fixes a bug with localized sites and Schedulers. 
When using a Locale like (for example) ES or IT, dates coming from TaskSchedulerController are breaking our DatePicker component. 
I also cleaned up React code a bit. 

As a next step, in a separate pull request, we should format DatePicker value according to current site's Locale.

Commit:
Scheduler date/time format issue with localized sites.

daguiler commented on Extensions/Settings/Dnn.PersonaBar.TaskScheduler/TaskScheduler.Web/src/components/scheduler/schedulerEditor/index.jsx:
```diff
@@ -387,6 +350,7 @@ class SchedulerEditor extends Component {
 
             return ( 
                 <div className="scheduler-setting-editor">
+                    CIAO
```
```suggestion
```

daguiler reviewed (changes requested):
Build is breaking with
```
D:\a\1\s\Extensions\Settings\Dnn.PersonaBar.TaskScheduler\TaskScheduler.Web\src\components\scheduler\schedulerEditor\index.jsx
##[error]279(0,0): Error : JSX attributes must only be assigned a non-empty expression
```
Also please apply suggestion.

Commit:
Code cleanup

valadas commented:
I am not able to reproduce the original issue on 9.3.0.106 , I tried adding fr-ca, es-es, it-it. I switched to each of those languages on the site before using the scheduler module, I tried setting the user preferred locale to each. In all cases the calendar in the scheduler was working.

 @mean2me  Could you provide steps to reproduce please.

daguiler commented:
Here's how I reproduced on 9.3.0-alpha.103.7992:
1) Specified a DD/MM/YYYY locale (es-ES) as the default of the site
2) Opened Scheduler and saw all dates were displayed correctly in DD/MM/YYYY format
3) I created a test task, leaving start date null
4) Clicked on Edit Task icon. No issues, I was able to edit the newly created task, BUT datepicker showed English format (MM/DD/YYYY). This didn't happen in 9.2.2.
5) I specified 12/12/2018 as start date. Saved. Tried to edit again. No issues.
6) Then I specified 12/**13**/2018. Saved. Tried to edit again. This time, the screen went blank and got several javascript errors in the browser console:
```
Uncaught TypeError: Cannot read property 'length' of undefined
    at Object.t.getWeekArray (export-bundle.js?cdv=52:12)
    at t.value (export-bundle.js?cdv=52:98)
    at ua (export-bundle.js?cdv=52:90)
    at la (export-bundle.js?cdv=52:90)
    at pa (export-bundle.js?cdv=52:90)
    at qa (export-bundle.js?cdv=52:90)
    at Qa (export-bundle.js?cdv=52:90)
    at xi (export-bundle.js?cdv=52:90)
    at Pi (export-bundle.js?cdv=52:90)
    at Di (export-bundle.js?cdv=52:90)
```
Error occurs at /DesktopModules/admin/Dnn.PersonaBar/scripts/exports/export-bundle.js?cdv=52

valadas commented:
Thank you, I was missing the default site language part. I will re-review tonight.

daguiler commented on Extensions/Settings/Dnn.PersonaBar.TaskScheduler/TaskScheduler.Web/src/components/scheduler/schedulerEditor/index.jsx:
```diff
@@ -314,6 +276,7 @@ class SchedulerEditor extends Component {
                     <Label label={resx.get("plScheduleStartDate")} style={{ margin: "0 0 5px 0" }} />
                     <DatePicker
                         date={this.getValue("ScheduleStartDate")}
+                        mode={}
```
This line is generating error
```
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: D:\a\1\s\Extensions\Settings\Dnn.PersonaBar.TaskScheduler\TaskScheduler.Web\src\components\scheduler\schedulerEditor\index.jsx:
JSX attributes must only be assigned a non-empty expression (279:29)

  277 |                     <DatePicker
  278 |                         date={this.getValue("ScheduleStartDate")}
> 279 |                         mode={}
      |                              ^
  280 |                         updateDate={this.onSettingChange.bind(this, "ScheduleStartDate")}
  281 |                         isDateRange={false}
  282 |                         hasTimePicker={true}
```

valadas committed:
Removes no longer used state

Prevents a warning on build.

valadas committed:
Removes unused mode={}

this props is used to set the icon, if not supplied the default icon shows (not startDate or endDate) which is fine. So it is not needed and causes a compilation error. Removing it completelly looks like the correct thing to do here.

valadas commented:
I added 2 commits that address the build issue and one build warning, tested locally and it looks good to me, but the CI did not fire with my commits, not sure why?...

valadas reviewed (approved):
Tested locally, fixes the issue and cleans up the code, I added 2 commits to fix the build error and one build warning. Other than that, looks good to me.

daguiler reviewed (approved):
Thanks @valadas!

valadas commented:
You are welcome @daguiler
